app/scripts/nav.js

```diff
@@ -27,18 +29,33 @@ $(function() {
 
     lastScrollTop = currentScrollTop;
 
-    if (isOnTop(currentScrollTop) && !nav.hasClass(onTopClass) ||
-        (!isOnTop(currentScrollTop) && nav.hasClass(onTopClass)))
-      toggleNavOnTop();
+    if (nav.hasClass(onTopClassIndex)) {
+      last_class = onTopClassIndex;
```

Comment:
I don't get this. Why set the `last_class` conditionally only to use it in a conditional after? Why not place the conditional code directly here?

The way I see it, you can declare a function to contain the second conditional instead:

```javascript
// ...
if (nav.hasClass(onTopClassIndex)) {
  updateNavOnTop(onTopClassIndex);
} else {
  updateNavOnTop(onTopClass);
}

// ...

function updateNavOnTop(selected_class) {
  const outdated = (isOnTop(currentScrollTop) && !nav.hasClass(onTopClassIndex)) ||
    (!isOnTop(currentScrollTop) && nav.hasClass(onTopClassIndex));
  if (outdated) {
    toggleNavOnTop(onTopClassIndex);
  }
}